doc/languages-frameworks/go.section.md

```diff
@@ -62,6 +62,56 @@ The following is an example expression using `buildGoModule`:
 }
 ```
 
+### Obtaining and overriding `vendorHash` for `buildGoModule` {#buildGoModule-vendorHash}
+
+`nix-prefetch` can be used to obtain the actual hash. The following command gets the value of `vendorHash` for package `pet`:
+
+```sh
+cd path/to/nixpkgs
+nix-prefetch -E "{ hash }: ((import ./. { }).my-package.overrideAttrs (_: { vendorHash = hash; })).go-modules"
+```
+
+To obtain the hash without external tools, set `vendorHash = lib.fakeHash;` and run the build. ([more details here](#sec-source-hashes)).
+
+`vendorHash` can be overridden with `overrideAttrs`. Override the above example like this:
+
+```nix
+{
+  pet_0_4_0 = pet.overrideAttrs (finalAttrs: previousAttrs: {
+    version = "0.4.0";
+    src = fetchFromGitHub {
+      inherit (previousAttrs.src) owner repo;
+      rev = "v${finalAttrs.version}";
+      hash = "sha256-gVTpzmXekQxGMucDKskGi+e+34nJwwsXwvQTjRO6Gdg=";
+    };
+    vendorHash = "sha256-dUvp7FEW09V0xMuhewPGw3TuAic/sD7xyXEYviZ2Ivs=";
+  });
+}
+```
+
+### Overriding `goModules` (#buildGoModule-goModules-override)
+
+In the past, overriding `<pkg>.goModules` was virtually impossible (see [#86349](https://github.com/NixOS/nixpkgs/issues/86349)). Now, it is possible to override its `outputHash` and unwrapping/patching attributes by overriding `vendorHash` and corresponding attributes in the primary derivation.
+
+Overriding `<pkg>.goModules` by calling `goModules.overrideAttrs` is still unsupported. This is because when the final state of attribute `vendorHash` becomes `null`, `goModules` becomes `""`. Alternatively, the primary derivation provides an overridable `passthru.overrideModAttrs` to store the attribute overlay implicitly taken by `goModules.overrideAttrs`.
```

Comment:
Maybe there is no gain when mentioning our ugly history? That is mentioned in the release notes already.

```suggestion
Overriding `<pkg>.goModules` by calling `goModules.overrideAttrs` is unsupported, but it is possible to override the `outputHash` and unwrapping/patching go modules by overriding `vendorHash` and corresponding attributes in the primary derivation. Alternatively, the primary derivation provides a `passthru.overrideModAttrs` function to store the attribute overlay implicitly taken by `goModules.overrideAttrs`. Here's an example usage of `overrideModAttrs`:
```

I deleted the sentence:

> This is because when the final state of attribute `vendorHash` becomes `null`, `goModules` becomes `""`. 

Because it is not phrased well and I think it is a not important implementation detail.

And I also replaced `unwrapping/patching attributes` to `unwrapping/patching go modules`, as that seems less ambiguous? I'm not sure about this detail of the sentence.

Reply:
By "unwrapping/patching attributes," I mean `prePatch`, `patches`, `patchFlags`, `postPatch`, `preBuild`, `sourceRoot`, and `setSourceRoot`. Do you know how to rephrase this to make it less ambiguous?